### PR TITLE
Update variant related drops

### PR DESCRIPTION
**Pull request by geoffreymm**
We've add/complete docs for:

- AccommodationVariantDrop
- ExperienceVariantDrop
- VariantDrop
- DepositDrop

We've also moved the `accommodation` merchandise field to a new file:
`docs/reference/objects/product/accommodation_merchandise.md` to remove the ambiguity with the `Accommodation` product type.
We've moved the `AvailabilityDay` from `product/accommodation` to:
docs/reference/objects/product/variant/accommodation_variant/availabilty_day.md

### CODE REVIEW

**Review by eballantine — APPROVED** [review]

File: docs/reference/objects/product/variant/index.md [hunk 1]
```diff
@@ -12,53 +12,137 @@ docs/reference/objects/product/variant/experience_variant/index.md %}) and
...
+{% raw %}
+```liquid
+{{assign price = variant.prices | first}}
+{{price.fractional | money}}
+```
+{% endraw %}
+
+# extra.product
+
+Returns the [product]({% link docs/reference/objects/product/index.md %}) this extra belongs to.
```
Comment (ianmooney) [inline docs/reference/objects/product/variant/index.md:125]: Change "extra" to "variant":

```
# variant.product

Returns the [product]({% link docs/reference/objects/product/index.md %}) this variant belongs to.
```

**Review by ianmooney — COMMENTED** [review]
Nice work, Geoff 🚀 

Left a few comments in there with small things I noticed.

**Comment by geoffreymm** [comment]
[Update drops documentation on easol.dev](https://trello.com/c/glhZWKqz/53-update-drops-documentation-on-easoldev)

File: docs/reference/objects/product/variant/accommodation_variant/index.md [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+When using an `accommodation_variant` object this will inherit all methods of
+the [variant]({% link docs/reference/objects/product/variant/index.md %}) plus
+it has access to the following attributes:
+
+# variant.max_occupancy
+
+Returns the maximum number of guests for this accommodation.
+
+# variant.min_occupancy
```
[the remark this answers is not in the document]
Comment (geoffreymm) [reply docs/reference/objects/product/variant/accommodation_variant/index.md:16]: Good spot. We have it duplicated on the `AccommodationVariantDrop`, I'll refactor to remove it.

File: docs/reference/objects/product/variant/index.md [hunk 3]
```diff
@@ -2,10 +2,15 @@
 layout: default
```
[the remark this answers is not in the document]
Comment (geoffreymm) [reply docs/reference/objects/product/variant/index.md:2]: 🙈 thanks for spotting that, I should have tested it.

**Comment by geoffreymm** [comment]
@ianmooney thanks for the patience on reviewing this one, feel like I was a bit slack on some of it. 🙈 Can you take another look? It should be less painful this time.

File: docs/reference/objects/product/variant/index.md [hunk 4]
```diff
@@ -62,3 +71,8 @@ Returns the variants tagline
 # variant.type
 
 Returns the type of this variant, one of `"experience_variant"` or `"accommodation_variant"`.
+Check what other methods you can call on this drop according to their type,
+[`experience_variant`]({% link
+docs/reference/objects/product/variant/experience_variant/index.md %}),
+[`accommodation_variant`]({% link
+docs/reference/objects/product/variant/accommodation_variant/index.md %}).
```
Comment (ianmooney) [inline docs/reference/objects/product/variant/index.md:154]: Maybe worth have these links titelized, e.g. 

```
[Experience Variant]({% link
docs/reference/objects/product/variant/experience_variant/index.md %}),
[Accommodation Variant]({% link
docs/reference/objects/product/variant/accommodation_variant/index.md %})
```

**Review by ianmooney — APPROVED** [review]
Nice one - looks good!

No worries, there were lots of changes going on.

**Pull request merged**